str.go

```diff
@@ -0,0 +1,85 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+var (
+	ErrorBadEpoch = errors.New("[merkletree] Bad STR Epoch. STR's epoch must be nonzero")
+)
+
+// SignedTreeRoot represents a signed tree root, which is generated
+// at the beginning of every epoch.
+// Signed tree roots contain the current root node,
+// the current and previous epochs, the hash of the
+// previous STR, and its signature.
+// STR should be final
+type SignedTreeRoot struct {
+	tree        *MerkleTree
+	epoch       uint64
+	prevEpoch   uint64
+	prevStrHash []byte
+	sig         []byte
+	policies    Policies
+}
+
+func NewSTR(key crypto.SigningKey, policies Policies, m *MerkleTree, epoch uint64, prevHash []byte) *SignedTreeRoot {
+	if epoch < 1 {
+		panic(ErrorBadEpoch)
+	}
+	str := &SignedTreeRoot{
+		tree:        m,
+		epoch:       epoch,
+		prevEpoch:   epoch - 1,
+		prevStrHash: prevHash,
+		policies:    policies,
+	}
+	bytesPreSig := str.getSTRBytesForSig()
+	str.sig = crypto.Sign(key, bytesPreSig)
+	return str
+}
+
+func (str *SignedTreeRoot) getSTRBytesForSig() []byte {
+	var strBytes []byte
+	strBytes = append(strBytes, util.ULongToBytes(str.epoch)...)     // t - epoch number
+	strBytes = append(strBytes, util.ULongToBytes(str.prevEpoch)...) // t_prev - previous epoch number
```

Comment:
since `prevEpoch` is `uint64` type. so I think it would be
```
if str.epoch > 0 {
        strBytes = append(strBytes, util.ULongToBytes(str.prevEpoch)...)
}